### PR TITLE
Roundup of minor code issues from the initial OpenGov PR

### PR DESCRIPTION
**Pull request by akhavich**
## Description
Fixed many small issues identified in the review of the OpenGov PR, including:
- simplifying code by removing unnecessary interfaces
- removing unused returns
- moving code and classes to more appropriate places in the project and namespace hierarchy
- simplifying Blazor/HTML markup
- converting a polling task to use an event listener
- and more

## How Has This Been Tested?
Manually verified functionality after each change.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Unit test (new or update to tests)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (change that would cause documentation to be updated)
- [ ] Configuration change (change that would affect the configuration of the application)

## Checklist:

- [x] PR is submitted to the correct branch `develop`.
- [x] Code follows the code style of this project.
- [ ] Changes are covered by Specflow tests and all new or existing tests are passing.
- [ ] Any new or updated translatable strings have been added to the localization files.
- [ ] Necessary and relevant documentation has been created or updated.
- [x] There are either no changes to the application's configuration or the necessary changes in the infrastructure repository are included in a separate PR.

### CODE REVIEW

File: Portal/src/Datahub.Portal/Pages/ShareSpatial.razor [hunk 1]
```diff
@@ -32,7 +33,7 @@
             {
                 <AeCard>
                     <AeTypography Variant="h3">@StepTitle(SpatialObjectShareStatus.FillApprovalForm)</AeTypography>
```
Comment (yjmrobert) [inline Portal/src/Datahub.Portal/Pages/ShareSpatial.razor]: Just a heads up there's some Ae components here, can remove later

File: Portal/src/Datahub.Portal/Startup.cs [hunk 2]
```diff
@@ -468,7 +468,6 @@ private void ConfigureCoreDatahubServices(IServiceCollection services)
 
         services.AddSingleton<ServiceAuthManager>();
 
-        services.AddSingleton<ICKANServiceFactory, CKANServiceFactory>();
```
Comment (yjmrobert) [inline Portal/src/Datahub.Portal/Startup.cs:448]: I don't think we can remove this if it's still being used by the `OpenDataService`. What do you think of updating that service to instantiate the factory instead?

**Review by yjmrobert — COMMENTED** [review]
Nice work, I'm not sure it will work without that factory as part of the DI but let me know if I'm wrong and I'll merge it!

File: Portal/src/Datahub.Portal/Pages/ShareSpatial.razor (hunk 1 above)
Comment (akhavich) [reply Portal/src/Datahub.Portal/Pages/ShareSpatial.razor]: I believe this is an unused legacy component - likely why it hadn't been updated with MudBlazor in the first place.

File: Portal/src/Datahub.Portal/Startup.cs (hunk 2 above)
Comment (akhavich) [reply Portal/src/Datahub.Portal/Startup.cs:448]: Done. It's legacy so it'll be removed eventually anyway, but for now that'll do.

**Pull request merged**